File: solutions/day1.ts
import fs from "node:fs";

const raw = fs.readFileSync("./input.txt", "utf-8");
const location = raw.split("\n").map((pair) => pair.split("   "));
const locations = location[0]
  .map((_, colIndex) => location.map((row) => parseInt(row[colIndex])))
  .map((list) => list.toSorted());

// Part 1
const sumPart1 = location.reduce(
  (acc, _, i) => acc + Math.abs(locations[0][i] - locations[1][i]),
  0
);
console.log("Part 1:", sumPart1);

// Part 2
const sumPart2 = location.reduce(
  (acc, _, i) =>
    acc +
    locations[1].filter((location) => location === locations[0][i]).length *
      locations[0][i],
  0
);
console.log("Part 2:", sumPart2);

File: solutions/python/day1.py
with open("./input.txt", "r") as txt:
  loc: str = txt.read().rstrip()
  txt.close()

location = [_.split('   ') for _ in loc.split('\n')]
locations = [list(map(lambda x: int(x[_]), location)) for _ in range(2)]

# Part 1
locations[0].sort()
locations[1].sort()
sumPart1 = 0
for i in range(len(location)):
  sumPart1 += abs(locations[0][i] - locations[1][i])
print("Part 1:", sumPart1)

# Part 2
sumPart2 = 0
for i in range(len(location)):
  sumPart2 += locations[1].count(locations[0][i]) * locations[0][i]
print("Part 2:", sumPart2)
